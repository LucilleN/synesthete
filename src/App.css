.App {
  /* margin-left: 5rem;
  margin-right: 5rem; */
  width: 100%;
  /* padding-left: 5rem;
  padding-right: 5rem; */
}
