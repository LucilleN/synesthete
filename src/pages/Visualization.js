import React, { useState, useEffect, useRef } from 'react'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'
import RecommendationButton from '../components/RecommendationButton'
import StartButton from '../components/StartButton'
import ErrorDialog from '../components/ErrorDialog'

import { Redirect } from 'react-router-dom'

import { getRecommendation } from '../api'
import { loadMusic, performAudioFeaturesQuery } from './visualizationUtilities'
import { defaultErrorText } from '../App'

export const styles = theme => ({
  root: {
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  fileInput: {
    zIndex: 3,
    color: 'white'
  },
  urlInput: {
    zIndex: 3,
    color: theme.palette.dark.pink,
    width: 250,
    margin: 15
  },
  urlButton: {
    zIndex: 3
  },
  recommendationButton: {
    zIndex: 10,
    marginTop: 200
  },
  canvas: {
    position: 'fixed',
    top: '60px',
    left: '0px',
    width: '100%',
    height: 'calc(100%-60px)',
    background: 'black',
    zIndex: 0
  },
  audio:{
    position: 'fixed',
    bottom: '20px',
    width: '90vw',
    height: '35px',
    zIndex: 3,
    opacity: 0.1,
    '&:hover': {
       opacity: 0.8,
    }
  },
  titleBar: {
    width: '100%',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-between'
  },
  title: {
    zIndex: 4,
    color: theme.palette.light.pink,
    fontSize: '3rem',
    textAlign: 'center',
    marginTop: 5
  },
  subtitle: {
    zIndex: 4,
    color: theme.palette.dark.pink,
    fontSize: '1.75rem',
    textAlign: 'center',
    marginBottom: 8,
    marginTop: -32
  },
  songInfoContainer: {
    zIndex: 4,
    width: '100%'
  },
  songInfo: {
    zIndex: 4,
    color: theme.palette.light.pink,
    fontSize: '0.75rem',
    textAlign: 'left',
    width: '100%',
    marginLeft: 8
  },
  loadOptionButton: {
    zIndex: 4,
    color: theme.palette.white,
    fontSize: '1rem',
    background: theme.palette.dark.pink,
    height: '60px',
    width: '250px',
    borderRadius: '30px',
    border: 'none',
    margin: '10px 20px',
    '&:hover': {
       background: theme.palette.white,
       color: theme.palette.dark.pink
    }
  },
  buttonBar: {
    width: '100%',
    display: 'flex',
    flexDirection: 'row',
    justifyContent: 'space-evenly'
  },
  button: {
    zIndex: 4,
    color: theme.palette.white,
    fontSize: '1rem',
    background: theme.palette.dark.purple,
    height: '60px',
    width: '250px',
    borderRadius: '30px',
    border: 'none',
    margin: '10px 20px',
    textTransform: 'none',
    '&:hover': {
       background: theme.palette.white,
       color: theme.palette.dark.purple
    }
  }
})

const Visualization = props => {
  const { classes } = props
  const { trackObject } = props.location.state

  if (!trackObject) {
    console.log("no track object")
    setError(defaultErrorText)
  }

  // May need this in the future to allow users to access a song's visualization just by
  // clicking on a saved link rather than going through the search every time, but this
  // would require another API call to get the trackObject based on the songID, so I
  // chose not to implement it for this project.
  // const { songID } = props.match.params

  const audioRef = useRef(null)
  const canvasRef = useRef(null)

  const [error, setError] = useState(null)
  const [audioFeatures, setAudioFeatures] = useState(null)
  const [recommendedSong, setRecommendedSong] = useState(null)

  const [existingAudioNode, setAudioNode] = useState(null)
  const [existingAnalyser, setAnalyser] = useState(null)
  const [existingContext, setContext] = useState(null)

  const [canLoadMusic, setCanLoadMusic] = useState(false)

  useEffect(() => {
    if (trackObject) {
      performAudioFeaturesQuery({
        setError,
        songID: trackObject.id,
        setAudioFeatures,
        defaultErrorText
      })
    }
  }, [trackObject])

  useEffect(() => {
    setCanLoadMusic(true)
  }, [audioFeatures])

  useEffect(() => {
    if (canLoadMusic && audioFeatures) {
      loadMusic({
        audioRef,
        canvasRef,
        srcUrl: trackObject.preview_url,
        existingContext,
        setContext,
        existingAudioNode,
        setAudioNode,
        existingAnalyser,
        setAnalyser,
        setError,
        audioFeatures
      })
    }
    setCanLoadMusic(false)
  }, [canLoadMusic])

  const performRecommendationQuery = async event => {
    event.preventDefault()
    setError(null)

    let currentRecommendation

    try {
      const result = await getRecommendation({
        limit: 1,
        seed_tracks: `${trackObject.id}`,
        seed_artists: `${trackObject.artists[0].id}`,
        target_energy: `${audioFeatures.energy}`,
        target_valence: `${audioFeatures.valence}`,
        target_acousticness: `${audioFeatures.acousticness}`,
        target_key: `${audioFeatures.key}`
      })
      currentRecommendation = result.tracks[0]

      const isSameSong = (track1, track2) => {
        return (
          track1.id === track2.id &&
          track1.artists[0].id === track2.artists[0].id &&
          track1.title === track2.title
        )
      }

      // Recommendation might be a song with a null preview_url; if so, recommend another one
      // Recommendation might also be the current song itself; if so, recommend another
      while (!currentRecommendation.preview_url || isSameSong(trackObject, currentRecommendation)) {
        console.log("currentTrack's id", trackObject.id)
        console.log("currentTrack's preview_url", trackObject.preview_url)
        console.log("currentRecommendation.id", currentRecommendation.id)
        const result = await getRecommendation({
          limit: 3,
          seed_tracks: `${currentRecommendation.id}`,
          // seed_artists: `${currentRecommendation.artists[0].id}`
        })
        currentRecommendation = result.tracks[0]
        console.log("got a new recommendation with id", currentRecommendation.id)
        console.log("new recommendation's preview url", currentRecommendation.preview_url)
      }

      setRecommendedSong(result.tracks[0])

    } catch (error) {
      setError(defaultErrorText)
    }
  }

  return (
    <div className={classes.root}>
      <canvas ref={canvasRef} id="canvas" width="300" height="300" className={classes.canvas}></canvas>
      <audio ref={audioRef} id="audio" crossOrigin="anonymous" controls className={classes.audio}></audio>
      <div className={classes.titleBar}>
        <StartButton href="/search" text="< new search"/>
        <Typography className={classes.title}>
          Visualization:<br></br>
        </Typography>
        <RecommendationButton className={classes.recommendationButton} handleClick={performRecommendationQuery}/>
      </div>
      <Typography className={classes.subtitle}>
        "{trackObject.name}" by {trackObject.artists[0].name}
      </Typography>
      {recommendedSong && (
        <Redirect to={{
            pathname: `/visualization/${recommendedSong.id}`,
            state: { trackObject: recommendedSong }
          }}
        />
      )}
      {error && (
        <div id="error">
          <ErrorDialog error={error} errorSubtitle={"The Visualizer doesn't work when it's been loaded externally. Try refreshing the page or starting a new search."}/>
        </div>
      )}
    </div>
  )
}

export default withStyles(styles)(Visualization)
