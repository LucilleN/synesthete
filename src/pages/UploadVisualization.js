import React, { useRef } from 'react'
import Typography from '@material-ui/core/Typography'
import { withStyles } from '@material-ui/core/styles'
import { styles } from './Visualization'

/*
const styles = theme => ({
  root: {
    // background: 'black',
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center'
  },
  fileInput: {
    // position: 'fixed',
    // top: '10px',
    // left: '10px',
    zIndex: 3,
    color: 'white'
  },
  canvas: {
    // position: 'fixed',
    // top: '0px',
    // left: '0px',
    width: '90vw',
    height: '70vh',
    // background: theme.palette.dark.purple
    background: 'black'
  },
  audio:{
    // position: 'fixed',
    // left: '10px',
    // bottom: '10px',
    width: 'calc(100%-25px)',
    width: '90vw',
    // background: theme.palette.dark.purple,
    // borderRadius: 0,
    zIndex: 3
  },
  background: {
    width: '100%',
    // height: '100%',
    height: 'calc(100% - 60px)',
    // width: '90vw',
    // height: '70vh',
    position: 'absolute',
    // top: '0px',
    // left: '0px',
    background: 'linear-gradient(transparent, #00035f, transparent)',
    backgroundSize: '100% 7px',
    // animation: 'bg 1s infinite linear',
    zIndex: 2,
    opacity: 0.3
  },
  '@keyframes bg': {
    from: { backgroundPosition: '0 0' },
    to: { backgroundPosition: '8px 8px' }
  },
  title: {
    color: theme.palette.white,
    fontSize: '2rem',
    zIndex: 4
  }
})
*/

const UploadVisualization = props => {
  const { classes } = props

  const audioRef = useRef(null)
  const canvasRef = useRef(null)
  const fileRef = useRef(null)

  /*-------------------------------------------
                LOAD MUSIC FILE
  -------------------------------------------*/
  const loadMusicFile = () => {
    const audio = audioRef.current

    const file = fileRef.current
    const files = file.files
    audio.src = URL.createObjectURL(files[0])
    // audio.crossOrigin = "anonymous";
    // audio.src = "https://p.scdn.co/mp3-preview/1c0da00b5c95a1a6c9dfc05b14a1a628a6e0ad73?cid=159ac88b1c534ed7ae41602f1e558a49"
    // audio.src = "https://www.soundhelix.com/examples/mp3/SoundHelix-Song-8.mp3"
    console.log("audio.src: " + audio.src)

    const canvas = canvasRef.current;
    canvas.width = window.innerWidth;
    canvas.height = window.innerHeight;
    const ctx = canvas.getContext("2d");

    const context = new AudioContext(); // (Interface) Audio-processing graph
    let src = context.createMediaElementSource(audio); // Give audio context an audio source
    const analyser = context.createAnalyser(); // Create an analyser for the audio context

    src.connect(analyser); // Connects the audio context source to the analyser
    analyser.connect(context.destination); // End destination of an audio graph in a given context
    analyser.fftSize = 16384;

    const bufferLength = analyser.frequencyBinCount;
    const dataArray = new Uint8Array(bufferLength);

    const WIDTH = canvas.width;
    const HEIGHT = canvas.height;

    const barWidth = (WIDTH / bufferLength) * 40;

    let barHeight;
    let x = 0;

    function renderFrame() {
      requestAnimationFrame(renderFrame); // Takes callback function to invoke before rendering

      x = 0;

      analyser.getByteFrequencyData(dataArray);

      ctx.fillStyle = "rgba(0,0,0,0.1)"; // Clears canvas before rendering bars (black with opacity 0.2)
      ctx.fillRect(0, 0, WIDTH, HEIGHT); // Fade effect, set opacity to 1 for sharper rendering of bars

      let r, g, b, a;
      let bars = 120 // Set total number of bars you want per frame

      const colors = [
        [179, 0, 54], // red
        [255, 60, 0], // red-orange
        [255, 123, 0], // orange
        [255, 178, 0], // honey yellow
        [255, 243, 163], // light yellow
        [203, 247, 57], // yellow-green
        [25, 191, 0], // green
        [0, 161, 155], // blue-green
        [0, 181, 169], // green-blue
        [0, 113, 212], // blue
        [39, 16, 173], // indigo
        [75, 0, 145], // dark purple
        [145, 10, 145], // magenta
      ]

      function getColor(index, offset) {
        let newIndex = index + offset
        if (newIndex >= colors.length) {
          newIndex -= colors.length
        }
        if (newIndex < 0) {
          newIndex += colors.length
        }
        return colors[newIndex]
      }

      for (let i = 0; i < bars; i++) {
        barHeight = (dataArray[i] * 2.5);
        const barHeightMax = 255 * 2.5;
        const multiple = 6
        // offset = -10 // Lord Huron, the Night We Met (key: A major)
        let offset = -7 // Kina Grannis, Iris (key: F# Major)

        if (i < 1 * multiple){ // red
          [r, g, b] = getColor(0, offset)
        } else if (i < 2 * multiple){ // red-orange
          [r, g, b] = getColor(1, offset)
        } else if (i < 3 * multiple){ // orange
          [r, g, b] = getColor(2, offset)
        } else if (i < 4 * multiple){ // yellow
          [r, g, b] = getColor(3, offset)
        } else if (i < 5 * multiple){ // light yellow
          [r, g, b] = getColor(4, offset)
        } else if (i < 6 * multiple){ // yellow-green
          [r, g, b] = getColor(5, offset)
        } else if (i < 7 * multiple){ // green
          [r, g, b] = getColor(6, offset)
        } else if (i < 8 * multiple){ // blue-green
          [r, g, b] = getColor(7, offset)
        } else if (i < 9 * multiple){ // green-blue
          [r, g, b] = getColor(8, offset)
        } else if (i < 10 * multiple){ // dark blue
          [r, g, b] = getColor(9, offset)
        } else if (i < 11 * multiple){ // indigo
          [r, g, b] = getColor(10, offset)
        } else if (i < 12 * multiple){ // dark purple
          [r, g, b] = getColor(11, offset)
        } else { // magenta
          [r, g, b] = getColor(12, offset)
        }

        a = (barHeight / barHeightMax) ** 3;

        ctx.fillStyle = `rgba(${r},${g},${b},${a})`;
        ctx.fillRect(x, (HEIGHT - barHeight), barWidth, barHeight);

        x += barWidth + 10 // Gives 10px space between each bar
      }
    }

    audio.play();
    renderFrame();
  }

  return (
    <div className={classes.root}>
      <Typography className={classes.title}>
        Upload A Song to Visualize
      </Typography>
      <input ref={fileRef} type="file" id="file-input" accept="audio/*,video/*,image/*" className={classes.fileInput} onChange={loadMusicFile}/>
      <canvas ref={canvasRef} id="canvas" width="300" height="300" className={classes.canvas}></canvas>
      <audio ref={audioRef} id="audio" controls className={classes.audio}></audio>
      {/* <div id="background" className={classes.background}></div> */}
    </div>
  )
}

export default withStyles(styles)(UploadVisualization)
