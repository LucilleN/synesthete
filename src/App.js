import React, { useEffect } from 'react'
import { Route, BrowserRouter as Router, Switch } from 'react-router-dom'

import { MuiThemeProvider, createMuiTheme } from '@material-ui/core/styles'
import { withStyles } from '@material-ui/core/styles'

import Homepage from './pages/Homepage'
import About from './pages/About'
import Search from './pages/Search'
import UploadVisualization from './pages/UploadVisualization'
import Visualization from './pages/Visualization'
import MenuBar from './components/MenuBar'

import { apiHost } from './api'

export const theme = createMuiTheme({
  palette: {
    primary: {
      main: '#c23063', // magenta
      dark: '#180142' // dark purple
    },
    type: 'dark',
    dark: {
      pink: '#bf265c',
      purple: '#1c004f',
    },
    medium: {
      purple: '#5a0b66',
    },
    light: {
      pink: '#db91aa',
      purple: '#ac91db'
    },
    white: '#FFFFFF',
  },
  typography: {
    fontFamily: ['Dosis', 'Roboto', 'Open Sans', '"Helvetica Neue"', 'Arial', 'sans-serif'].join(',')
  }
})

export const defaultErrorText = 'Sorry, but something went wrong.'

const styles = () => ({
  root: {
    width: '100vw',
    height: '100%',
    minHeight: '100vh',
    background: `linear-gradient(0deg, ${theme.palette.dark.pink} 0%, ${theme.palette.dark.purple} 80%)`
  }
})

const routes = (
  <Router>
    <div>
      <MenuBar />
      <Switch>
        <Route exact path="/" component={Homepage} />
        <Route exact path="/about" component={About} />
        <Route exact path="/search" component={Search} />
        <Route exact path="/upload" component={UploadVisualization} />
        <Route exact path="/visualization" component={Visualization} />
        <Route exact path="/visualization/:songID" component={Visualization} />
      </Switch>
    </div>
  </Router>
)

const App = (props) => {

  useEffect(() => apiHost('https://api.spotify.com/v1/'))

  const { classes } = props

  return (
    <div className={classes.root}>
      <MuiThemeProvider theme={theme}>
        {routes}
      </MuiThemeProvider>
    </div>
  )
}

export default withStyles(styles)(App)
