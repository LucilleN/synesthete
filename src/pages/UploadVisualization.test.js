import React from 'react'
import ReactDOM from 'react-dom'
import TestRenderer from 'react-test-renderer'
import ReactTestUtils from 'react-dom/test-utils'
import { MuiThemeProvider } from '@material-ui/core/styles'

import sinon from 'sinon'

import { theme } from '../App'
import UploadVisualization from './UploadVisualization'
import * as visualizationModule from './Visualization'

const sampleURL = "https://p.scdn.co/mp3-preview/1c0da00b5c95a1a6c9dfc05b14a1a628a6e0ad73?cid=159ac88b1c534ed7ae41602f1e558a49"

let component;
beforeEach(() => {
  component = TestRenderer.create(
    <MuiThemeProvider theme={theme}>
      <UploadVisualization />
    </MuiThemeProvider>
  )
})

describe('initial state', () => {
  let div
  beforeEach(() => {
    div = document.createElement('div')
    ReactTestUtils.act(() => {
      ReactDOM.render(
        <MuiThemeProvider theme={theme}>
          <UploadVisualization />
        </MuiThemeProvider>
      , div)
    })
  })

  it('should display a title and buttons to reset and navigate to home', () => {
    // console.log("UploadVisualization test, component", component)
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should start with two buttons for load options', () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    const uploadButton = div.querySelector('#upload-file')
    expect(uploadButton).not.toBeNull()
    const loadUrlButton = div.querySelector('#load-url')
    expect(loadUrlButton).not.toBeNull()
  })

  it('should not display the file input or the text input', () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    const fileInput = div.querySelector('#file-input')
    expect(fileInput).toBeNull()
    const textInput = div.querySelector('#text-input')
    expect(textInput).toBeNull()
  })


})

describe('clicking on the Upload File button', () => {
  let div
  beforeEach(() => {
    div = document.createElement('div')
    ReactTestUtils.act(() => {
      ReactDOM.render(
        <MuiThemeProvider theme={theme}>
          <UploadVisualization />
        </MuiThemeProvider>
      , div)
    })
    const uploadButton = div.querySelector('#upload-file')
    ReactTestUtils.act(() => {
      ReactTestUtils.Simulate.click(uploadButton)
    })
  })

  afterEach(() => ReactDOM.unmountComponentAtNode(div))

  it('should show the file input element', () => {
    const fileInput = div.querySelector('#file-input')
    expect(fileInput).not.toBeNull()
  })

  it('should not show the load option buttons', () => {
    const uploadButton = div.querySelector('#upload-file')
    expect(uploadButton).toBeNull()
    const loadUrlButton = div.querySelector('#load-url')
    expect(loadUrlButton).toBeNull()
  })

})

describe('clicking on the Load URL button', () => {
  let div
  beforeEach(() => {
    div = document.createElement('div')
    ReactTestUtils.act(() => {
      ReactDOM.render(
        <MuiThemeProvider theme={theme}>
          <UploadVisualization />
        </MuiThemeProvider>
      , div)
    })
    const loadUrlButton = div.querySelector('#load-url')
    ReactTestUtils.act(() => {
      ReactTestUtils.Simulate.click(loadUrlButton)
    })
  })

  afterEach(() => ReactDOM.unmountComponentAtNode(div))

  it('should show the text input element', () => {
    const textInput = div.querySelector('#text-input')
    expect(textInput).not.toBeNull()
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should not show the load option buttons', () => {
    const uploadButton = div.querySelector('#upload-file')
    expect(uploadButton).toBeNull()
    const loadUrlButton = div.querySelector('#load-url')
    expect(loadUrlButton).toBeNull()
  })

  describe('load button', () => {
    it('should be disabled when the text field is blank', () => {
      const loadButton = div.querySelector('#load-button')
      expect(loadButton.disabled).toBe(true)
    })

    it('should be enabled when the text field is not blank', () => {
      const urlTextInput = div.querySelector('#text-input')
      ReactTestUtils.act(() => {
        urlTextInput.value = 'a mock URL'
        ReactTestUtils.Simulate.change(urlTextInput)
      })

      const loadButton = div.querySelector('#load-button')
      expect(loadButton.disabled).toBe(false)
    })

  })

})

// ----------------------------------------------------------

describe('loading a URL', () => {
  let div
  beforeEach(() => {
    sinon.stub(visualizationModule, 'loadMusic')

    div = document.createElement('div')
    ReactTestUtils.act(() => {
      ReactDOM.render(
        <MuiThemeProvider theme={theme}>
          <UploadVisualization />
        </MuiThemeProvider>
      , div)
    })
    const loadUrlButton = div.querySelector('#load-url')
    ReactTestUtils.act(() => {
      ReactTestUtils.Simulate.click(loadUrlButton)
    })
    const textInput = div.querySelector('#text-input')
    const loadButton = div.querySelector('#load-button')
    ReactTestUtils.act(() => {
      textInput.value = sampleURL
      ReactTestUtils.Simulate.change(textInput)
    })
    ReactTestUtils.act(() => {
      ReactTestUtils.Simulate.click(loadButton) //maybe put this in above act
    })


  })

  afterEach(() => {
    ReactDOM.unmountComponentAtNode(div)
    visualizationModule.loadMusic.restore()
  })

  it('should trigger the music to load when the load button is clicked', () => {
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
    expect(visualizationModule.loadMusic.calledOnce).toBe(true)
  })

})
