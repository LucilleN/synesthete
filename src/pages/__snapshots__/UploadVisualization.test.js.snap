// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`clicking on the Load URL button should show the text input element 1`] = `
<div
  className="UploadVisualization-root-1"
>
  <div
    className="UploadVisualization-titleBar-8"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiFab-root UploadVisualization-button-15"
      href="/"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiFab-label"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          &lt; back to home
        </p>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
    <p
      className="MuiTypography-root UploadVisualization-title-9 MuiTypography-body1"
    >
      Load a Song to Visualize
    </p>
    <button
      className="UploadVisualization-button-15"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Reset
      </p>
    </button>
  </div>
  <canvas
    className="UploadVisualization-canvas-6"
    height="300"
    id="canvas"
    width="300"
  />
  <audio
    className="UploadVisualization-audio-7"
    controls={true}
    crossOrigin="anonymous"
    id="audio"
  />
  <div
    className="UploadVisualization-buttonBar-14"
  >
    <button
      className="UploadVisualization-loadOptionButton-13"
      id="upload-file"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Upload a file from your computer
      </p>
    </button>
    <button
      className="UploadVisualization-loadOptionButton-13"
      id="load-url"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Enter the URL of an audio file
      </p>
    </button>
  </div>
</div>
`;

exports[`initial state should display a title and buttons to reset and navigate to home 1`] = `
<div
  className="UploadVisualization-root-1"
>
  <div
    className="UploadVisualization-titleBar-8"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiFab-root UploadVisualization-button-15"
      href="/"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiFab-label"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          &lt; back to home
        </p>
      </span>
    </a>
    <p
      className="MuiTypography-root UploadVisualization-title-9 MuiTypography-body1"
    >
      Load a Song to Visualize
    </p>
    <button
      className="UploadVisualization-button-15"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Reset
      </p>
    </button>
  </div>
  <canvas
    className="UploadVisualization-canvas-6"
    height="300"
    id="canvas"
    width="300"
  />
  <audio
    className="UploadVisualization-audio-7"
    controls={true}
    crossOrigin="anonymous"
    id="audio"
  />
  <div
    className="UploadVisualization-buttonBar-14"
  >
    <button
      className="UploadVisualization-loadOptionButton-13"
      id="upload-file"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Upload a file from your computer
      </p>
    </button>
    <button
      className="UploadVisualization-loadOptionButton-13"
      id="load-url"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Enter the URL of an audio file
      </p>
    </button>
  </div>
</div>
`;

exports[`initial state should not display the file input or the text input 1`] = `
<div
  className="UploadVisualization-root-1"
>
  <div
    className="UploadVisualization-titleBar-8"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiFab-root UploadVisualization-button-15"
      href="/"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiFab-label"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          &lt; back to home
        </p>
      </span>
    </a>
    <p
      className="MuiTypography-root UploadVisualization-title-9 MuiTypography-body1"
    >
      Load a Song to Visualize
    </p>
    <button
      className="UploadVisualization-button-15"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Reset
      </p>
    </button>
  </div>
  <canvas
    className="UploadVisualization-canvas-6"
    height="300"
    id="canvas"
    width="300"
  />
  <audio
    className="UploadVisualization-audio-7"
    controls={true}
    crossOrigin="anonymous"
    id="audio"
  />
  <div
    className="UploadVisualization-buttonBar-14"
  >
    <button
      className="UploadVisualization-loadOptionButton-13"
      id="upload-file"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Upload a file from your computer
      </p>
    </button>
    <button
      className="UploadVisualization-loadOptionButton-13"
      id="load-url"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Enter the URL of an audio file
      </p>
    </button>
  </div>
</div>
`;

exports[`initial state should start with two buttons for load options 1`] = `
<div
  className="UploadVisualization-root-1"
>
  <div
    className="UploadVisualization-titleBar-8"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiFab-root UploadVisualization-button-15"
      href="/"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiFab-label"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          &lt; back to home
        </p>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
    <p
      className="MuiTypography-root UploadVisualization-title-9 MuiTypography-body1"
    >
      Load a Song to Visualize
    </p>
    <button
      className="UploadVisualization-button-15"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Reset
      </p>
    </button>
  </div>
  <canvas
    className="UploadVisualization-canvas-6"
    height="300"
    id="canvas"
    width="300"
  />
  <audio
    className="UploadVisualization-audio-7"
    controls={true}
    crossOrigin="anonymous"
    id="audio"
  />
  <div
    className="UploadVisualization-buttonBar-14"
  >
    <button
      className="UploadVisualization-loadOptionButton-13"
      id="upload-file"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Upload a file from your computer
      </p>
    </button>
    <button
      className="UploadVisualization-loadOptionButton-13"
      id="load-url"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Enter the URL of an audio file
      </p>
    </button>
  </div>
</div>
`;

exports[`loading a URL should trigger the music to load when the load button is clicked 1`] = `
<div
  className="UploadVisualization-root-1"
>
  <div
    className="UploadVisualization-titleBar-8"
  >
    <a
      aria-disabled={false}
      className="MuiButtonBase-root MuiFab-root UploadVisualization-button-15"
      href="/"
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
    >
      <span
        className="MuiFab-label"
      >
        <p
          className="MuiTypography-root MuiTypography-body1"
        >
          &lt; back to home
        </p>
      </span>
      <span
        className="MuiTouchRipple-root"
      />
    </a>
    <p
      className="MuiTypography-root UploadVisualization-title-9 MuiTypography-body1"
    >
      Load a Song to Visualize
    </p>
    <button
      className="UploadVisualization-button-15"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Reset
      </p>
    </button>
  </div>
  <canvas
    className="UploadVisualization-canvas-6"
    height="300"
    id="canvas"
    width="300"
  />
  <audio
    className="UploadVisualization-audio-7"
    controls={true}
    crossOrigin="anonymous"
    id="audio"
  />
  <div
    className="UploadVisualization-buttonBar-14"
  >
    <button
      className="UploadVisualization-loadOptionButton-13"
      id="upload-file"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Upload a file from your computer
      </p>
    </button>
    <button
      className="UploadVisualization-loadOptionButton-13"
      id="load-url"
      onClick={[Function]}
    >
      <p
        className="MuiTypography-root MuiTypography-body1"
      >
        Enter the URL of an audio file
      </p>
    </button>
  </div>
</div>
`;
